tests/pdm/backend/hooks/version/test_scm.py

```diff
@@ -0,0 +1,288 @@
+            files: The files to include when creating the commit
+        """
+        ...
+
+    @abstractmethod
+    def tag(self, name: str):
+        """Create a tag
+
+        Args:
+            name: The name for the provided tag
+        """
+        ...
+
+    @property
+    @abstractmethod
+    def current_hash(self) -> str: ...
+
+
+class GitScm(Scm):
+    def __init__(self, cmd: Path, cwd: Path) -> None:
+        super().__init__(cmd, cwd)
+        self.run("config", "commit.gpgsign", "false")
+
+    def _init(self):
+        self.run("init")
+
+    def commit(self, commit_message: str, files: list[Path]):
+        self.run("add", *files)
+        self.run("commit", "-m", commit_message)
+
+    def tag(self, name: str):
+        self.run("tag", "-am", "some tag", name)
+
+    @property
+    def current_hash(self) -> str:
+        return "g" + self.run("rev-parse", "--short", "HEAD").strip()
+
+
+class HgScm(Scm):
+    def _init(self):
+        self.run("init")
+
+    def commit(self, commit_message: str, files: list[Path]):
+        self.run("add", *files)
+        self.run("commit", "-m", commit_message, *files)
+
+    def tag(self, name: str):
+        self.run("tag", name)
+
+    @property
+    def current_hash(self) -> str:
+        return self.run("id", "-i").strip()
+
+
+@pytest.fixture
+def scm_dir() -> Iterable[Path]:
+    d = Path(tempfile.mkdtemp())
+
+    yield d
+
+    shutil.rmtree(d)
+
+
+@pytest.fixture
+def git(scm_dir: Path) -> GitScm:
+    git = shutil.which("git")
+    assert git is not None, "Cannot find git in path"
+
+    scm = GitScm(Path(git), scm_dir)
+
+    return scm
+
+
+@pytest.fixture
+def hg(scm_dir: Path) -> HgScm:
+    hg = shutil.which("hg")
+    assert hg is not None, "Cannot find git in path"
+
+    scm = HgScm(Path(hg), scm_dir)
+
+    return scm
+
+
+@pytest.fixture(params=[lazy_fixture("git"), lazy_fixture("hg")])
+def scm(request: pytest.FixtureRequest, scm_dir: Path) -> Scm:
+    scm = cast(Scm, request.param)
+
+    file_path = scm_dir / "test.txt"
+    with open(file_path, "w") as f:
+        f.write("a\n")
+
+    scm.commit("Add a", [file_path])
+
+    return scm
+
+
+def test__get_version_from_scm__returns_tag_if_method_unspecified(
+    scm_dir: Path, scm: Scm
+):
+    expected_tag = "0.2.52"
+    scm.tag(expected_tag)
+
+    version = get_version_from_scm(scm_dir)
+
+    assert version is not None
+    assert version == expected_tag
+
+
+def test__get_version_from_scm__adds_details_if_project_is_dirty(
+    scm_dir: Path, scm: Scm
+):
+    expected_tag = "0.2.52"
+    file_path = scm_dir / "some_file.txt"
+
+    with open(file_path, "w") as f:
+        f.write("having fun\n")
+    scm.commit("some commit", [file_path])
+    scm.tag(expected_tag)
+    with open(file_path, "a") as f:
+        f.write("having fun 2\n")
+
+    version = get_version_from_scm(scm_dir)
+
+    assert version is not None
+    assert version == f"{expected_tag}+d{datetime.now(tz.UTC).strftime('%Y%m%d')}"
```

Comment:
```suggestion
    assert version == f"{expected_tag}+d{datetime.now(timezone.utc).strftime('%Y%m%d')}"
```

And `from datetime import timezone`
This works for all Python versions supported

Reply:
Done.